src/main/java/uk/ac/cam/cl/dtg/segue/api/QuestionFacade.java

```diff
@@ -196,13 +199,26 @@ public Response answerQuestion(@Context final HttpServletRequest request,
                 }
             } else {
                 try {
+                    // Monitor misuse on a per-question per-anonymous user basis:
                     misuseMonitor.notifyEvent(((AnonymousUserDTO) currentUser).getSessionId() + "|" + questionId,
                             AnonQuestionAttemptMisuseHandler.class.toString());
                 } catch (SegueResourceMisuseException e) {
                     this.getLogManager().logEvent(currentUser, request, QUESTION_ATTEMPT_RATE_LIMITED, response.getEntity());
                     String message = "You have made too many attempts at this question part. Please log in or try again later.";
                     return SegueErrorResponse.getRateThrottledResponse(message);
                 }
+                try {
```

Comment:
Perhaps in this case we should care more if the user is persistently getting the question wrong? This may protect us from blocking genuine users. i.e. reset the counter for the IP address if the question is correct? It does allow bad users to hide more easily.